src/main/java/org/apache/commons/lang3/time/TimingRecordNode.java

```diff
@@ -0,0 +1,222 @@
+     * @param tags the tags to associate with this timing
+     * @throws IllegalArgumentException if the timingName is null or empty.
+     */
+    public TimingRecordNode(String parentTimingPath, String timingName, String... tags) {
+        if (StringUtils.isEmpty(timingName)) {
+            throw new IllegalArgumentException("Argument name is missing");
+        }
+        this.timingName = timingName;
+
+        if (StringUtils.isNotEmpty(parentTimingPath)) {
+            this.parentTimingPath = parentTimingPath;
+        }
+
+        this.tags = tags;
+    }
+
+    /**
+     * Returns the node's parent's path.
+     * The parent node path may be null
+     *
+     * @return the parent node path
+     */
+    public String getParentPath() {
+        return parentTimingPath;
+    }
+
+    /**
+     * Returns the node's timing name.
+     *
+     * @return the node timing name
+     */
+    public String getTimingName() {
+        return timingName;
+    }
+
+    /**
+     * Return if the node's StopWatch is running.
+     *
+     * @return true if it is running, false if not
+     */
+    public boolean isRunning() {
+        return stopWatch.isStarted();
+    }
+
+    /**
+     * Starts the StopWatch.
+     */
+    public void start() {
+        if (!stopWatch.isStarted()) {
+            stopWatch.start();
+        }
+    }
+
+    /**
+     * <p>
+     * Stops the StopWatch.
+     * </p>
+     * <p>
+     * If this node has running children, an {@code IllegalStateException} will result.
+     * </p>
+     *
+     * @throws IllegalStateException if stop is called on a node with running children
+     */
+    public void stop() {
+        for (TimingRecordNode child : children) {
+            if (child.isRunning()) {
+                throw new IllegalStateException("Cannot stop a timing with running children");
+            }
+        }
+        stopWatch.stop();
+    }
+
+    /**
+     * Returns the {@code StopWatch} for this node.
+     *
+     * @return {@code StopWatch}
+     */
+    public StopWatch getStopWatch() {
+        return stopWatch;
+    }
+
+    /**
+     * The tags associated with this timing.
+     *
+     * @return tags array
+     */
+    public String[] getTags() {
+        return tags == null ? new String[]{} : ArrayUtils.clone(tags);
+    }
+
+
+    /**
+     * Returns the node's path, made up by combining it's parent's name and the node's name.
+     * The path is '/' delimited.
+     * <p>
+     * If the parent path is null, then the name only is returned.
+     * </p>
+     *
+     * @return the path as String
+     */
+    public String getPath() {
+        if (parentTimingPath == null) {
+            return timingName;
+        }
+        return String.format(PATH_FMT, parentTimingPath, timingName);
+    }
+
+    /**
+     * Returns the child nodes of this node.
+     *
+     * @return Iterable of the child nodes.
+     */
+    public Iterable<TimingRecordNode> getChildren() {
+        return children;
```

Comment:
Couldn't we return an read-only copy of the children here?

Reply:
done.